hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/SqlConnector.java

```diff
@@ -229,6 +231,31 @@ default Vertex fullScanReader(
         throw new UnsupportedOperationException("Full scan not supported for " + typeName());
     }
 
+    /**
+     * Returns a supplier for a source vertex reading the index according to
+     * the {@code filter} with {@code projection} and {@code predicate}.
+     * The output type of the source is Object[].
+     *
+     * @param table      the table object
+     * @param predicate  SQL expression to filter the rows
+     * @param projection the list of field names to return
+     * @param filter     internal index filter
+     * @return The DAG Vertex handling the reading
+     */
+    @Nonnull
+    default Vertex indexScanReader(
```

Comment:
I think it's quite unlikely that any other connector, except `IMap` one, will implement this method (because of `IndexFilter`).